WordPress/src/main/java/org/xmlrpc/android/XMLRPCUtils.java

```diff
@@ -313,13 +313,25 @@ private static String discoverSelfHostedXmlrpcUrl(String siteUrl, String httpUse
         // add the url as provided by the user
         urlsToTry.add(siteUrl);
 
-        // add a sanitized version of the url
-        final String sanitizedURL = sanitizeSiteUrl(siteUrl);
-        if (!urlsToTry.contains(sanitizedURL)) {
-            urlsToTry.add(sanitizedURL);
+        // add a sanitized version of the http url (if the user didn't specify it)
+        final String sanitizedHttpsURL = sanitizeSiteUrl(siteUrl, true);
+        if (!urlsToTry.contains(sanitizedHttpsURL)) {
+            urlsToTry.add(sanitizedHttpsURL);
+        }
+
+        String appendedXmlrpcUrl = appendXMLRPCPath(sanitizedHttpsURL);
+        if (!urlsToTry.contains(appendedXmlrpcUrl)) {
+            appendedXmlrpcUrl += "?rsd";
```

Comment:
The discovery procedure tries to parse HTML document to get the rsd URL. But it doesn't add it if it's not found. 

But you're right this is useless since the code below doesn't try to parse the result document as an RSD file (this was also wrong before the patch).

It also doesn't try to POST on it, so `urlsToTry` with URL ending by `/xmlrpc.php` are useless too.